### ISSUE TITLE
Multiple incomplete skeletons assigned to one animal in bottom-up model inference

### ISSUE THREAD

**Issue by littleshibeta** [talmolab/sleap#2529]
Hi,

I am new here. I would like to share my problem about bottom-up model inference and if anyone could help or would like to share useful information will be great appreciated!

Recently I am analyzing my videos with sleap. My videos are gray-scale, about 1000x1000 pixels in size, and there are multiple mice in each video. The number of animals is not fixed, like, one mouse might be added to the recording site at the middle. there are at most 4 mice in one frame.

I designed a skeleton containing 7 bodyparts, including: nose, head, chest, belly, butt, and left & right ear. They are in an arrow shape, like the picture below, so the model could distinguish the heading of the animal. 

<img width="281" height="407" alt="Image" src="https://github.com/user-attachments/assets/abd2ec5a-540b-4594-adb2-79720a81afc4" />

The video is a little bit of, not clear, but I can distinguish most bodyparts with naked eye.

I labeled 600 frames and trained a  bottom-up model, batch_size was set to 8, and other hyper parameters were left as default. Training took around one hour for 40 epochs on a 4080. The validation loss was around 9 x e^-5. Then, The whole video was analyzed with this model.

On most frames, especially when animals are far away from each other, the model performed pretty well. However, at some frames, Multiple skeletons were assigned to single animal, which was pretty strange and made me unable to perform precise tracking. 

<img width="471" height="579" alt="Image" src="https://github.com/user-attachments/assets/c8217b87-b9b9-42bd-9366-7d601ca77f31" />

<img width="935" height="426" alt="Image" src="https://github.com/user-attachments/assets/cbcfded5-6b1e-45f2-8e1d-e832c7439ebb" />

<img width="630" height="599" alt="Image" src="https://github.com/user-attachments/assets/f1992e7f-752c-428a-82a6-67f0a170799d" />

I tried labeling more frames, and the model could recognize bodyparts more precisely, but the situation did not seem to be better. There were still a lot of bugs like the pictures above.

The only solution I could imagine is that, combining the centroid model with this bottom-up model. However the GUI has no such options and I dont know if this is possible, or if there are better solutions.

Also, it would be very difficult for me to improve the video quality. I think this is the best I can provide by now.

Please help me solve this! I am sincerely grateful to anyone who would like to help. 

p.s I hope you could understand my words, since I am not from a English-speaking country.

Thanks again!

**Comment by littleshibeta** [comment]
More details: I am using the SLEAP version 1.4.1a2. Hope this could be useful.

I tried top-down model and it seemed to perform better at assigning skeletons to individuals. However, top-down model cannot learn the structure of the skeleton(e.g., ears are always closer to nose, not the butt ), and always mess up the predicted instance. I don't know if there are better ways to train, or infer, with top-down model.

And, another supplementary information: when I label the animal, I tend to label all the bodyparts, even they are not shown in the current frame. I would roll back and forward for a few frames to predict the actual position of covered bodyparts(for example, one ear might be covered in this frame and displayed in the next frame), because I need the information of all bodyparts to run further behavior analysis, and I want the model to be able to predict the position of covered bodyparts as well. Maybe this labeling method is no good for accuracy?

**Comment by littleshibeta** [comment]
Update on 1/12/2026: After labeled over 1700 frames and 4000 instances, the model seems to perform pretty well on most frames. I know this is a silly solution, and I still wonder if there are clever ways to achieve the same performance. Thanks!

**Comment by LeahBlankenship** [comment]
I am also having this same issue in very similar videos, using SLEAP 1.5.2. We have tried limiting the number of max instances, increasing and decreasing the sigma for nodes and edges, and changing the input scaling. The model is labeling the points with relatively good accuracy but it's doing the exact same thing shown above. When we limit max instances to 2, it simply only puts one of the two parts of the skeleton on the animal. It would be great to find a better fix for this than simply labeling extremely high numbers of frames. What is confusing to me is that we had started training this model on SLEAP 1.4 and did not have this problem. Then when we transferred the file to SLEAP 1.5 on a new computer, the model performance got significantly worse and as we have been re-training it to get it back to where it was, it has developed this issue. Any solutions would be much appreciated!
